src/main/java/net/sf/jabref/logic/importer/fetcher/zbMATH.java

```diff
@@ -0,0 +1,138 @@
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Objects;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import net.sf.jabref.logic.cleanup.FieldFormatterCleanup;
+import net.sf.jabref.logic.cleanup.MoveFieldCleanup;
+import net.sf.jabref.logic.formatter.bibtexfields.ClearFormatter;
+import net.sf.jabref.logic.formatter.bibtexfields.RemoveBracesFormatter;
+import net.sf.jabref.logic.help.HelpFile;
+import net.sf.jabref.logic.importer.FetcherException;
+import net.sf.jabref.logic.importer.ImportFormatPreferences;
+import net.sf.jabref.logic.importer.Parser;
+import net.sf.jabref.logic.importer.SearchBasedParserFetcher;
+import net.sf.jabref.logic.importer.fileformat.BibtexParser;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.FieldName;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.client.utils.URIBuilder;
+
+/**
+ * Fetches data from the Zentralblatt Math (https://www.zbmath.org/)
+ */
+public class zbMATH implements SearchBasedParserFetcher {
+
+    private static final Log LOGGER = LogFactory.getLog(zbMATH.class);
+
+    private final ImportFormatPreferences preferences;
+
+    public zbMATH(ImportFormatPreferences preferences) {
+        this.preferences = Objects.requireNonNull(preferences);
+    }
+
+    @Override
+    public String getName() {
+        return "zbMATH";
+    }
+
+    @Override
+    public HelpFile getHelpPage() {
+        return null;
+    }
+
+    /**
+     * TODO: Implement EntryBasedParserFetcher
+     * We use the zbMATH Citation matcher (https://www.zbmath.org/citationmatching/)
+     * instead of the usual search since this tool is optimized for finding a publication based on partial information.
+     */
+    /*
+    @Override
+    public URL getURLForEntry(BibEntry entry) throws URISyntaxException, MalformedURLException, FetcherException {
+        // Example: https://zbmath.org/citationmatching/match?q=Ratiu
+    }
+    */
+
+    @Override
+    public URL getURLForQuery(String query) throws URISyntaxException, MalformedURLException, FetcherException {
+        URIBuilder uriBuilder = new URIBuilder("https://zbmath.org/bibtexoutput/");
+        uriBuilder.addParameter("q", query); // search all fields
+        uriBuilder.addParameter("start", "0"); // start index
+        uriBuilder.addParameter("count", "200"); // should return up to 200 items (instead of default 100)
+
+        fixSSLVerification();
+
+        return uriBuilder.build().toURL();
+    }
+
+    /**
+     * Older java VMs does not automatically trust the zbMATH certificate. In this case the following exception is thrown:
```

Comment:
Can we install this hack only if JRE is 8 and < 101? I feel uncomfortable to install that fix globally.

Alternatively, can this handler be uninstalled after using this fetcher? Maybe when calling `getParser`?